src/Sulu/Bundle/AdminBundle/Resources/js/containers/Form/fields/Selection.js

```diff
@@ -40,12 +42,53 @@ class Selection extends React.Component<Props> {
             fieldTypeOptions: {
                 resource_key: resourceKey,
             },
+            formInspector,
+            schemaOptions: {
+                request_parameters: {
+                    value: requestParameters = [],
+                } = {},
+                resource_store_properties_to_request: {
+                    value: resourceStorePropertiesToRequest = [],
+                } = {},
+            },
         } = this.props;
 
         if (!resourceKey) {
             throw new Error('The selection field needs a "resource_key" option to work properly');
         }
 
+        if (!Array.isArray(requestParameters)) {
+            throw new Error('The "request_parameters" schemaOption must be an array!');
+        }
+
+        if (!Array.isArray(resourceStorePropertiesToRequest)) {
+            throw new Error('The "resource_store_properties_to_request" schemaOption must be an array!');
+        }
+
+        this.requestOptions = this.buildRequestOptions(
+            requestParameters,
+            resourceStorePropertiesToRequest,
+            formInspector
+        );
+
+        // update requestOptions observable if one of the "resource_store_properties_to_request" properties is changed
+        const observedDataPaths = resourceStorePropertiesToRequest.map((property) => {
```

Comment:
Why don't you move that code into the finish handler? This way it won't work if the `resourceSTorePropertiesToRequest` change.